### PR TITLE
Merge pull request #1656 from flix-tech/asg-tags-resource-override

### PR DESCRIPTION
**Pull request by muaazsaleem**
Adding ability to override allocatable resources via ASG tags.

### CODE REVIEW

**Comment by muaazsaleem** [comment]
👍

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 1]
```diff
@@ -259,6 +259,14 @@ func (m *AwsManager) buildNodeFromTemplate(asg *asg, template *asgTemplate) (*ap
 		}
 	}
 
+	resourcesFromTags := extractAllocatableResourcesFromAsg(template.Tags)
+	if val, ok := resourcesFromTags["ephemeral-storage"]; ok {
+		node.Status.Capacity[apiv1.ResourceEphemeralStorage] = *val
+	}
+
+	// TODO: use proper allocatable!!
```
Comment (szuecs) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: What does it mean?

File: cluster-autoscaler/cloudprovider/aws/aws_manager_test.go [hunk 2]
```diff
@@ -48,6 +48,31 @@ func TestBuildGenericLabels(t *testing.T) {
...
+		},
+		{
+			Key:   aws.String("k8s.io/cluster-autoscaler/node-template/resources/memory"),
+			Value: aws.String("100M"),
+		},
+		{
+			Key:   aws.String("k8s.io/cluster-autoscaler/node-template/resources/ephemeral-storage"),
+			Value: aws.String("20G"),
+		},
+	}
```
Comment (szuecs) [inline cluster-autoscaler/cloudprovider/aws/aws_manager_test.go:65]: Did you considered using table tests for some different values for example for cpu case "2"?

**Comment by muaazsaleem** [comment]
@szuecs, this is a cherry-pick of https://github.com/kubernetes/autoscaler/pull/1656 so not really "my" code

**Comment by mikkeloscar** [comment]
@muaazsaleem I think you cherry-picked the "wrong" commit which is the merge commit. It doesn't correctly attribute the original author. This would be the correct commit: https://github.com/kubernetes/autoscaler/pull/1656/commits/f79a432693d1d4923103c97695ae8170049224cb

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 3]
```diff
@@ -259,6 +259,14 @@ func (m *AwsManager) buildNodeFromTemplate(asg *asg, template *asgTemplate) (*ap
...
+	node.Status.Allocatable = node.Status.Capacity
```
Comment (aermakov-zalando) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go]: Please drop this, we already have better logic above and this would just overwrite it.

**Comment by muaazsaleem** [comment]
@mikkeloscar thanks for the catch. Pushed the correct commit.

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 4]
```diff
@@ -259,6 +259,11 @@ func (m *AwsManager) buildNodeFromTemplate(asg *asg, template *asgTemplate) (*ap
 		}
 	}
 
+	resourcesFromTags := extractAllocatableResourcesFromAsg(template.Tags)
+	if val, ok := resourcesFromTags["ephemeral-storage"]; ok {
+		node.Status.Capacity[apiv1.ResourceEphemeralStorage] = *val
```
Comment (aermakov-zalando) [inline cluster-autoscaler/cloudprovider/aws/aws_manager.go:264]: Please update `Allocatable` as well (otherwise pods still won't be scheduled).

Comment (aermakov-zalando) [reply cluster-autoscaler/cloudprovider/aws/aws_manager.go:264]: Here's how it looks in the 1.14 branch: https://github.com/zalando-incubator/autoscaler/blob/zalando-cluster-autoscaler/cluster-autoscaler/cloudprovider/aws/aws_manager.go#L400

Comment (muaazsaleem) [reply cluster-autoscaler/cloudprovider/aws/aws_manager.go:264]: Done.

**Comment by muaazsaleem** [comment]
👍

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**